### DISCUSSION TITLE
[offtopic/docker] What server to use for local hosting?

### DISCUSSION THREAD

**Discussion by bergziege** [mkdocs/mkdocs#3506] [General]
Hi there,

I know it sound like this is covered in the documentation but I have a few questions on the general idea.
In the end I would like to selfhost some documentation internally on my home server (no domain name/certifacte needed).

What do I currently have:
- Server (A) with gitea (for the md files) and some more docker containers
- gitea actions runner on a separate machine (B)
- a gitea workflow (read: deployment pipeline) executing on B

I am currently on the search for a simple web server (running on A/Docker) where I can deploy the artifacts from the pipeline (on B) .

The problem is that I can only find servers like nginx that expect the static content to be in a certain directory when starting up. But I would like some server with integrated FTP/WebDAV/SSH/... so that I can upload directly from my gitea pipeline to the webserver.

Cheers and happy coding
debgz

**Comment by Andre601** [comment]
No idea if that helps at all, but codeberg made a github-pages like server software that integrates into gitea instanses by using its api...

https://codeberg.org/Codeberg/pages-server
